packages/components/inline-message/src/inline-message.ts

```diff
@@ -0,0 +1,129 @@
+ * @slot details - slot for more details of the inline-message like list of errors
+ * @slot icon - icon shown on the left side of the component
+ * @slot close-button - Closing button for the inline message
+ */
+@localized()
+export class InlineMessage extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-button': Button,
+      'sl-icon': Icon
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = [breakpoints, styles];
+
+  /** @private */
+  @query('.inline-message-wrapper') wrapper?: HTMLDivElement;
+
+  /** Determines whether closing button (default one) should be shown in the top right corner. */
+  @property({ type: Boolean, reflect: true }) dismissible = true;
+
+  /** Determines whether the icon should be shown on the left side of the component. */
+  @property({ type: Boolean, attribute: 'no-icon' }) noIcon?: boolean;
+
+  /** The status of the inline message.
+   * @type {'info' | 'success' | 'warning' | 'danger'} */
+  @property({ reflect: true }) status: InlineMessageStatus = 'info';
+
+  /** @private The name of the icon, depending on the status of the inline message. */
+  get iconName(): string {
+    switch (this.status) {
+      case 'info':
+        return 'info';
+      case 'success':
+        return 'circle-check-solid';
+      case 'warning':
+        return 'fas-octagon-exclamation';
+      case 'danger':
+        return 'triangle-exclamation-solid';
+      default:
+        return 'info';
+    }
+  }
+
+  override async connectedCallback(): Promise<void> {
+    super.connectedCallback();
+
+    if (this.status === 'danger' || this.status === 'warning') {
+      this.setAttribute('role', 'alert');
+    } else {
+      this.setAttribute('role', 'status');
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <div class="inline-message-wrapper" open>
+        <div class="content">
+          ${!this.noIcon
+            ? html`<slot name="icon" part="icon">
+                <sl-icon name=${this.iconName} size="md"></sl-icon>
+              </slot>`
+            : nothing}
+          <div class="content-details">
+            <slot name="title" part="title">
+              <slot></slot>
+            </slot>
+            <slot name="description" part="description"></slot>
+            <slot name="details" part="details"></slot>
+          </div>
+        </div>
+        ${this.dismissible
+          ? html`
+              <slot name="close-button" @click=${this.onClose}>
+                <sl-button fill="ghost" variant="default" size="sm" aria-label=${msg('Close')}>
+                  <sl-icon name="xmark"></sl-icon>
+                </sl-button>
+              </slot>
+            `
+          : nothing}
+      </div>
+    `;
+  }
+
+  onClose(): void {
+    this.wrapper?.removeAttribute('open');
+    this.#closeOnAnimationend();
+  }
+
+  #closeOnAnimationend(): void {
```

Comment:
Is there a reason why we can't have the `animationend` event handler directly defined on the `.wrapper` div in the `render()` method?